From 7042d7244d2ea298e6a08cb1e2bc4e42dd7242cc Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Sat, 30 Jun 2018 16:58:42 +0200
Subject: [PATCH 22/23] build: Disable backuptool for A/B on -user

* This requires hacking around neverallows which is currently
  out of the question

Change-Id: Ib273962344545fdf3027edc205f193d678c0d8b8
---
 core/Makefile | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/core/Makefile b/core/Makefile
index ffceef6d6..ad1f780d1 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -2820,7 +2820,15 @@ ifeq ($(WITH_GMS),true)
     $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := false
 else
 ifneq ($(LINEAGE_BUILD),)
-    $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
+    ifeq ($(TARGET_BUILD_VARIANT),user)
+        ifeq ($(AB_OTA_UPDATER),true)
+            $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := false
+        else
+            $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
+        endif
+    else
+        $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := true
+    endif
 else
     $(INTERNAL_OTA_PACKAGE_TARGET): backuptool := false
 endif
-- 
2.17.1

